cachi2/core/package_managers/generic/models.py

```diff
@@ -29,6 +39,16 @@ def no_empty_checksums(cls, value: dict[str, str]) -> dict[str, str]:
             raise ValueError("At least one checksum must be provided.")
         return value
 
+    @property
+    def formatted_checksums(self) -> List[ChecksumInfo]:
+        """Return the checksums as a list of ChecksumInfo objects."""
+        return [ChecksumInfo(algo, digest) for algo, digest in self.checksums.items()]
+
+    @property
+    def resolved_target(self) -> str:
+        """Return the computed target file of the artifact, either set, or taken from the download URL."""
+        return self.target if self.target else Path(str(self.download_url)).name
```

Comment:
> should be taken care of with an after model validator (as field validators must take attribute order validation into consideration and so are painful to deal with) OR maybe there's a callback for the Field descriptor that could compute the value but then again, I don't think there's a different primitive apart from model validators that would not need to care about other attributes not having been validated/set yet.

My first idea with this was to do this directly in the `Field` with a lambda, but that can't see other fields in the model.

Do I understand correctly that you're suggesting I use the `after` model validator to set the value of `target` based on the `download_url` ? I've seen that suggested, but it felt somewhat as misusing the validator to set something. Although I am happy to implement it, if that's the preferred way to do it.

Reply:
> Do I understand correctly that you're suggesting I use the after model validator to set the value of target based on the download_url ? I've seen that suggested, but it felt somewhat as misusing the validator to set something

Yes. I hear you, the only other way would be to deal with this from the caller, i.e. when the lockfile model is instantiated from YAML. When I look at the official docs https://docs.pydantic.dev/latest/concepts/validators/#using-validation-context-with-basemodel-initialization they use some overly complex mechanism to change mangle the value that the model is supposed to be instantiated with which gives me the confidence that setting reasonable default values is within the context of those validators since IIUC no other mechanism exists for that purpose in pydantic.